signing-scheme.md

```diff
@@ -0,0 +1,97 @@
+These approaches also can provide different feature sets than ones traditionally provided by standard X.509 PKI based signing.
+E.g. TUF is geared for software update systems and provides a freshness guarantee.
+
+Notary v2 will initially support X.509 PKI identity based signing, but provide the flexibility for additional systems through an abstraction called Signing Scheme.
+The Signing Scheme covers aspects of signature generation and verification, and formalizes the feature set (guarantees) provided by the signature produced using a signing scheme.
+Generally it covers the following aspects, but can be extended to other aspects as required by newer systems.
+
+*Signature creation*
+
+1. Supported entities that can generate signatures - typically an end user generates the signature, other models can be supported using signature scheme.
+1. Representation of identities - e.g. X.509 certificates are used to represent end users and entities which verify the end user’s authenticity (CAs).
+1. Signature schema - Defines the signed and unsigned attributes to be included in a signature envelope, and which attributes are required, optional and critical.
+
+*Signature verification*
+
+1. Mechanism to establish trust in end users and other entities (like CAs).
+1. Set of guarantees available to the verifier apart from integrity and authenticity.
+
+## Signing Scheme
+
+Notary v2 currently defines the following Signing Schemes.
+
+`notary.x509` - Defines a signing scheme that uses the traditional signing workflow in which an end user generates signatures using X.509 certificates.
+
+`notary.x509.signingAuthority` - Defines a signing scheme in which a `signing authority` generates signatures on behalf of an end user (the signature requestor) using X.509 certificates. 
+A trusted signing authority is defined as a third party service that is trusted both by the end user (the signature requestor) and verifying entity to generate signatures.
+Authorities trusted by users should have mechanisms for validating, logging, monitoring, and auditing access to their systems and publish incidence response plans. 
+A certificate authority (CA) will need to demonstrate only validated entities were issued an X.509 certificate that chains to their root, and a time stamp authority (TSA) will need to demonstrate the time-stamp signing keys were only used within their service.
+Similarly, a signing authority (SA) will need to demonstrate signing keys were only used within their service and only validated entities were allowed to generate signatures using the service.
+
+* A signature envelope can only specify one Signing Scheme
+* When Notary supports an additional Signing Scheme
+  * Existing signed artifacts MUST be resigned if they need to be verified using the new signing scheme defined verification process.
+  * Existing clients used by verifying entity MUST be updated to newer versions that support verifying signatures that use the new signing scheme.
```

Comment:
to support backward compatability existing client must ignore(not fail) for new signing scheme?

Reply:
Yeah, this sounds a bit scary. 
Can you actually ignore a new `signingScheme`? Are you saying there may be multiple signatures, and a client configured with an older signingScheme can pass? But, what if you're verifying a public cert that used the old, and new signingScheme? Would it pass, because it's ignored?

Reply:
Addressed during call, older clients will fail specific signatures which use newer or unknown signing scheme, but will continue to allow signatures with known signing schemes to validate, if there are multiple signatures associated with the artifact. I've updated language.